src/common/preferences/preferences.jsx

```diff
@@ -0,0 +1,618 @@
+		var currentProxyIdx = -1, currentHostIdx = -1;
+		if (event.target.value == '+') {
+			this.state.proxies.push({id: Date.now(), scheme: 'http://%h.example.com/%p', autoAssociate: true, 
+				hosts: ['www.example.com']});
+			currentProxyIdx = this.state.proxies.length-1;
+			currentHostIdx = 0;
+		} else if (event.target.value == '-') {
+			this.state.proxies.splice(this.state.currentProxyIdx, 1);
+			Zotero.Proxies.remove(this.state.currentProxy);
+		}
+		this.saveProxies(currentProxyIdx, currentHostIdx);
+	},
+	
+	handleSchemeChange: function(event) {
+		this.state.currentProxy.scheme = event.target.value;
+		this.saveProxies(this.state.currentProxyIdx);
+	},
+	
+	handleCheckboxChange: function(event) {
+		this.state.currentProxy[event.target.name] = event.target.checked;
+		this.saveProxies(this.state.currentProxyIdx)
+	},
+	
+	handleHostSelectChange: function(event) {
+		var currentHostIdx = -1;
+		let selected = Array.from(event.target.options).filter((o) => o.selected);
+		if (selected.length == 1) {
+			currentHostIdx = parseInt(selected[0].value);
+		}
+		this.setState({currentHostIdx: currentHostIdx});
+	},
+	
+	handleHostButtonClick: function(event) {
+		var currentHostIdx = -1;
+		if (event.target.value == '+') {
+			this.state.currentProxy.hosts.push('www.example.com');
+			currentHostIdx = this.state.currentProxy.hosts.length-1;
+		} else if (event.target.value == '-') {
+			this.state.currentProxy.hosts.splice(this.state.currentHostIdx, 1);
+		}
+		this.setState({currentHostIdx: currentHostIdx});
+		this.saveProxies(this.state.currentProxyIdx);
+	},
+	
+	handleHostnameChange: function(event) {
+		this.state.currentProxy.hosts[this.state.currentHostIdx] = event.target.value;
+		this.saveProxies(this.state.currentProxyIdx);
+	},
+	
+	render: function(){
+		var configuredProxies;
+		if (this.state.proxies.length) {
+			configuredProxies = this.state.proxies.map((proxy, i) => 
+				<option value={i} key={i}>{proxy.scheme}</option> 
+			);
+		} else {
+			configuredProxies = <option value={-1}>N/A</option>;
+		}
+		
+		var proxySettings = "";
+		if (this.props.transparent && this.state.currentProxy) {
+			let multiHost = this.state.currentProxy.scheme.indexOf('%h') != -1;
+			let configuredHosts = this.state.currentProxy.hosts.map((host, i) => 
+				<option value={i} key={i}>{host}</option>);
+				
+			proxySettings = (
+				<div className="group" style={{marginTop: "10px"}}>
+					<p style={{display: "flex", alignItems: "center", flexWrap: "wrap"}}>
+						<label><input type="checkbox" name="dotsToHyphens" onChange={this.handleCheckboxChange} checked={this.state.currentProxy.dotsToHyphens}/>&nbsp;Automatically convert hyphens to dots in proxied hostnames</label><br/>
+						<label style={{visibility: multiHost ? null : 'hidden'}}><input type="checkbox" name="autoAssociate" onChange={this.handleCheckboxChange} checked={this.state.currentProxy.autoAssociate}/>&nbsp;Automatically associate new hosts</label><br/>
```

Comment:
"Automatically associate new hosts" should be first, and the hyphens pref should probably be on a new line